zmq-lwt/src/deferred.ml

```diff
@@ -0,0 +1,24 @@
+open Lwt
+
+type 'a t = 'a Lwt.t
+let (>>=) = Lwt.(>>=)
+let return a = Lwt.return a
+let try_with f = Lwt_result.catch (f ())
+
+module Fd = struct
+  type nonrec 'a t' = 'a t
+  type t = Lwt_unix.file_descr
+
+  let create fd = Lwt_unix.of_unix_file_descr fd
+  let syscall_exn fd f = Lwt_unix.wrap_syscall Lwt_unix.Read fd f
+  let ready_to t kind =
```

Comment:
In the original implementation, this should just always return `Ready. 
https://github.com/hcarty/lwt-zmq/blob/master/src/lwt_zmq.ml#L21